src/complex/Utilities/DataArrayUtilities.hpp

```diff
@@ -904,4 +906,215 @@ class CopyTupleUsingIndexList
+  };
+
+  using Combine = TemplateTypeOptions<true, false>;
+  using Append = TemplateTypeOptions<false, true>;
+
+  template <class TemplateTypeOptions = Combine, class ParallelRunnerT, class... ArgsT>
+  static void RunParallel(IArray::ArrayType arrayType, IArray* destArray, ParallelRunnerT&& runner, ArgsT&&... args)
+  {
+    static_assert(!(TemplateTypeOptions::UsingCombine && TemplateTypeOptions::UsingAppend), "Cannot use both Append and Combine");
+    static_assert(!(!TemplateTypeOptions::UsingCombine && !TemplateTypeOptions::UsingAppend), "Cannot have Append and Combine false");
+
+    DataType dataType = DataType::int32;
+    if(arrayType == IArray::ArrayType::NeighborListArray)
+    {
+      dataType = dynamic_cast<INeighborList*>(destArray)->getDataType();
+    }
+    if(arrayType == IArray::ArrayType::DataArray)
+    {
+      dataType = dynamic_cast<IDataArray*>(destArray)->getDataType();
+      if(dataType == DataType::boolean)
+      {
+        if constexpr(TemplateTypeOptions::UsingCombine)
+        {
+          RunCombineBoolAppend(destArray, std::forward<ArgsT>(args)...);
+        }
+        if constexpr(TemplateTypeOptions::UsingAppend)
+        {
+          RunAppendBoolAppend(destArray, std::forward<ArgsT>(args)...);
+        }
+        return;
+      }
+    }
+
+    if constexpr(TemplateTypeOptions::UsingCombine)
+    {
+      ExecuteParallelFunction<CombineArrays, NoBooleanType>(dataType, std::forward<ParallelRunnerT>(runner), arrayType, destArray, std::forward<ArgsT>(args)...);
+    }
+    if constexpr(TemplateTypeOptions::UsingAppend)
+    {
+      ExecuteParallelFunction<AppendArray, NoBooleanType>(dataType, std::forward<ParallelRunnerT>(runner), arrayType, destArray, std::forward<ArgsT>(args)...);
+    }
+  }
+
+private:
+  template <class K>
+  static void AppendData(K* inputArray, K* destArray, usize offset)
+  {
+    const usize numElements = inputArray->getNumberOfTuples() * inputArray->getNumberOfComponents();
+    for(usize i = 0; i < numElements; ++i)
+    {
+      auto value = (*inputArray)[i]; // make sure we are getting a copy not a ref
+      (*destArray)[offset + i] = value;
+    }
+  }
+
+  template <typename T>
+  class AppendArray
+  {
+  public:
+    AppendArray(IArray::ArrayType arrayType, IArray* destCellArray, IArray* inputCellArray, usize tupleOffset)
+    : m_ArrayType(arrayType)
+    , m_InputCellArray(inputCellArray)
+    , m_DestCellArray(destCellArray)
+    , m_TupleOffset(tupleOffset)
+    {
+    }
+
+    ~AppendArray() = default;
+
+    AppendArray(const AppendArray&) = default;
+    AppendArray(AppendArray&&) noexcept = default;
+    AppendArray& operator=(const AppendArray&) = delete;
+    AppendArray& operator=(AppendArray&&) noexcept = delete;
+
+    void operator()() const
+    {
+      const usize offset = m_TupleOffset * m_DestCellArray->getNumberOfComponents();
+      if(m_ArrayType == IArray::ArrayType::NeighborListArray)
+      {
+        using nListT = NeighborList<T>;
```

Comment:
For type aliases, I would use something like `ArrayT` or `ArrayType` to be consistent with our other aliases.